versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java

```diff
@@ -123,13 +125,17 @@ Hash commit(
    * @param referenceHash The hash to use as a reference for conflict detection. If not present, do
    *     not perform conflict detection
    * @param sequenceToTransplant The sequence of hashes to transplant.
+   * @param updateCommitMetadata Function to allow resetting properties application due to merge.
    * @throws ReferenceConflictException if {@code referenceHash} values do not match the stored
    *     values for {@code branch}
    * @throws ReferenceNotFoundException if {@code branch} or if any of the hashes from {@code
    *     sequenceToTransplant} is not present in the store.
    */
   void transplant(
-      BranchName targetBranch, Optional<Hash> referenceHash, List<Hash> sequenceToTransplant)
+      BranchName targetBranch,
+      Optional<Hash> referenceHash,
+      List<Hash> sequenceToTransplant,
+      BiFunction<Serializer<METADATA>, ByteString, ByteString> updateCommitMetadata)
```

Comment:
This should be a `Function<METADATA, METADATA>`

Reply:
The serialiser is available with the StoreWorker, and appears to be available at an implementation layer of the version store storeWorker.getMetadataSerializer(). However, the logic to update CommitMeta has to be at TreeResource.
Hence, expecting a serialiser as input, and supplied by the VersionStore implementation.

Reply:
Serialization is a concern of the implementation, not the `VersionStore` interface.